### ISSUE TITLE
Feature: Slider to control thumbnail size

### ISSUE THREAD

**Issue by newyork2003** [files-community/Files#10492]
### What feature or improvement do you think would benefit Files?

Better fine tuning of folders and files viewing sizes

### Requirements

- [x] Add slider to control item sizes
    - [x] Details
    - [x] List
    - [x] Tiles (out of scope for now)
    - [x] Grid
    - [x] Columns
- [x] Add tooltips to slider
    - [x] Details
    - [x] List
    - [x] Tiles (out of scope for now)
    - [x] Grid
    - [x] Columns
- [x] Add icons to slider
    - [x] Details
    - [x] List
    - [x] Tiles (out of scope for now)
    - [x] Grid
    - [x] Columns
- [x] Update toolbar icon to use the smaller icon sizes
    - [x] Details
    - [x] List
    - [x] Tiles
    - [x] Grid
    - [x] Columns
- [ ] Keep scroll position when changing size (out of scope for now)
    - [x] Details
    - [ ] List
    - [ ] Tiles (out of scope for now)
    - [x] Grid
    - [x] Columns
 - [ ] Update thumbnail dimensions based on size (out of scope for now)
    - [x] Details
    - [x] List
    - [x] Tiles (out of scope for now)
    - [x] Grid
    - [x] Columns

### Files Version

2.3.12

### Windows Version

Windows 10 Pro Version	21H2 Installed on	‎25/‎08/‎2022 OS build	19044.2251 Experience	Windows Feature Experience Pack 120.2212.4180.0

**Comment by Yashirow** [comment]
It's great but with this implementation it's no longer possible to have two different grid sizes for two different folders. It automatically sync every sizes even if the "synchronize layout and sorting preferences between folders" option is unchecked.

**Comment by yaira2** [comment]
@Yashirow that's correct, we reworked the layout sizes to apply globally primarily so users can change layouts without losing their sizing preference but there were several other reasons as well.

**Comment by Yashirow** [comment]
@yaira2 okay thanks for the reply, I though it was a bug. I liked to have different sizes for different folders, mainly for image folders, which I set to maximum. But it's not a big loss, I think this feature is more important.

**Comment by yaira2** [comment]
We were a bit hesitant to make this breaking change for that reason, but when we gathered feedback, it turned out that most people tend to pick one size for all locations, and if the need comes up to change it, it's not a big deal to override id (there are also keyboard shortcuts for this). We'll continue monitoring feedback in this area and will make further adjustments as needed.

**Comment by Yashirow** [comment]
@yaira2 Okay. Thanks for you work !

**Comment by yaira2** [comment]
#15067 is requesting for this to also affect the font size, we aren't committing to this just yet but it's something we should consider.

**Comment by yaira2** [comment]
Updating the Tiles layout is out of scope for now. Everything else on our list has been implemented.

**Comment by yaira2** [comment]
The latest preview build (v3.8.13) completes this work.
